extensions/spring-xd-extension-reactor/src/main/java/org/springframework/xd/integration/reactor/net/ReactorPeerFactoryBean.java

```diff
@@ -0,0 +1,280 @@
+import java.util.HashMap;
+import java.util.Map;
+
+
+/**
+ * @author Stephane Maldini
+ */
+@SuppressWarnings({"unchecked", "rawtypes"})
+public class ReactorPeerFactoryBean<IN, OUT> implements FactoryBean<ReactorPeer<IN, OUT, ?>> {
+
+	private static final Map<String, Codec> DEFAULT_CODECS = new HashMap<String, Codec>();
+
+	static final String UDP  = "udp";
+	static final String TCP  = "tcp";
+	static final String HTTP = "http";
+
+	static {
+		DEFAULT_CODECS.put("bytes", StandardCodecs.BYTE_ARRAY_CODEC);
+		DEFAULT_CODECS.put("string", StandardCodecs.STRING_CODEC);
+		DEFAULT_CODECS.put("syslog", new SyslogCodec());
+	}
+
+	private final String transport;
+
+	private final Environment environment;
+
+	private final Map<String, Codec> codecs;
+
+	private String host;
+
+	private Dispatcher dispatcher;
+
+	private int port;
+
+	private String framing;
+
+	private int lengthFieldLength;
+
+	private Codec delegateCodec;
+
+	public ReactorPeerFactoryBean(Environment env) {
+		this(env, "tcp", null);
+	}
+
+	public ReactorPeerFactoryBean(Environment env, String transport) {
+		this(env, transport, null);
+	}
+
+	public ReactorPeerFactoryBean(Environment env, Map<String, Codec> codecs) {
+		this(env, "tcp", codecs);
+	}
+
+	public ReactorPeerFactoryBean(Environment env, String transport, Map<String, Codec> codecs) {
+		this.codecs = (null == codecs ? DEFAULT_CODECS : codecs);
+		this.transport = transport;
+		this.environment = (null == env ? Environment.initializeIfEmpty().assignErrorJournal() : env);
+	}
+
+	public ReactorPeerFactoryBean<IN, OUT> configure(URI uri) {
+		setHost(null != uri.getHost() ? uri.getHost() : "0.0.0.0");
+		setPort(uri.getPort() > 0 ? uri.getPort() : 3000);
+		setFraming(null != uri.getPath() ? uri.getPath().substring(1) : "linefeed");
+		this.delegateCodec = StandardCodecs.STRING_CODEC;
+
+		if (null != uri.getQuery()) {
+			String[] params = StringUtils.split(uri.getQuery(), "&");
+			if (null == params) {
+				params = new String[]{uri.getQuery()};
+			}
+			for (String pair : params) {
+				String[] parts = StringUtils.split(pair, "=");
+				if (parts.length > 1) {
+					if ("codec".equals(parts[0])) {
+						setCodec(parts[1]);
+					} else if ("dispatcher".equals(parts[0])) {
+						setDispatcher(parts[1]);
+					} else if ("lengthFieldLength".equals(parts[0])) {
+						setLengthFieldLength(Integer.parseInt(parts[1]));
+					}
+				}
+			}
+		}
+		return this;
+	}
+
+	/**
+	 * Set the name of the {@link reactor.core.Dispatcher} to use, which will be pulled from the current {@link
+	 * reactor.Environment}.
+	 *
+	 * @param dispatcher dispatcher name
+	 * @return {@literal this}
+	 */
+	public ReactorPeerFactoryBean<IN, OUT> setDispatcher(String dispatcher) {
+		Assert.notNull(environment, "Cannot lookup dispatcher from unassigned environment");
+		this.dispatcher = environment.getDispatcher(dispatcher);
+		return this;
+	}
+
+	public ReactorPeerFactoryBean<IN, OUT> setDispatcher(Dispatcher dispatcher) {
```

Comment:
I just ported the current code without any assumption on the feature but trying to make it work in the tests provided. But yeah, agree a lot of this is not used (was it ?) but I did want to just port 1-1 what you had, there is no addition everything has just been renamed and adapted.